packages/vite/src/node/plugins/css.ts

```diff
@@ -192,7 +192,15 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
           // record deps in the module graph so edits to @import css can trigger
           // main import to hot update
           const depModules = new Set(
-            [...deps].map((file) => moduleGraph.createFileOnlyEntry(file))
+            await Promise.all(
+              [...deps].map(async (file) =>
+                cssLangRE.test(file)
+                  ? moduleGraph.createFileOnlyEntry(file)
+                  : moduleGraph.ensureEntryFromUrl(
+                      await fileToUrl(file, config, this)
+                    )
+              )
```

Comment:
@paraboul I have been thinking about the use of `Promise.all` here and I think it is safer to avoid it and respect the order in which dependencies are discovered. I checked other parts in the code and If I understand correctly the module graph should be stable (at least this is useful for debugging, and it may actually end up affecting hashes)
```js
  const depModules = new Set()
  for (const dep of deps) {
    depModules.add( cssLangRE.test(dep) 
      ? moduleGraph.createFileOnlyEntry(file) )
      : await moduleGraph.ensureEntryFromUrl(
          await fileToUrl(file, config, this)
        )
    )
  }
```
Would you review and change this? I'll hold off this week release to wait for this PR